data/data.go

```diff
@@ -0,0 +1,135 @@
+package data
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/docker/engine/api/types"
+)
+
+// Source is an abstraction for a provider of task metadata + stats.
+// In practice, it'll be the ECS task metadata endpoint, but it can also be mocked out.
+type Source interface {
+	// Stats returns a map of DockerIDs to stats of the form returned by the Docker daemon's stats endpoint
+	Stats() (map[string]types.StatsJSON, error)
+	// Metadata retrieves the metadata for the task
+	Metadata() (TaskMetadata, error)
+}
+
+// TaskMetadata describes the response of `GET MetadataURI/task`
+// See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v3.html#task-metadata-endpoint-v3-response
+// and v4 is the same: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html#task-metadata-endpoint-v4-response
```

Comment:
That's Clever specific :) But it can be determined by this log https://github.com/Clever/ecs-task-metadata-exporter/blob/birthday/main.go#L93, anyway.